addons/misra.py

```diff
@@ -740,9 +740,10 @@ def get_num_significant_naming_chars(self, cfg):
 
     def misra_3_1(self, rawTokens):
         for token in rawTokens:
-            if token.str.startswith('/*') or token.str.startswith('//'):
+            starts_with_double_slash = token.str.startswith('//')
+            if token.str.startswith('/*') or starts_with_double_slash:
                 s = token.str.lstrip('/')
-                if '//' in s or '/*' in s:
+                if (not starts_with_double_slash and '//' in s) or '/*' in s:
```

Comment:
I want that the execution order is clarified.. if it means `(not (starts_with_double_slash and '//' in s))` or `((not starts_with_double_slash) and '//' in s)`

Reply:
Fixed